docs/how-to/reactive.md

```diff
@@ -0,0 +1,112 @@
+
+```shell
+juju switch machine-model
+juju deploy github-runner large --channel latest/stable ....
+juju deploy github-runner large-arm --channel latest/stable ....
+juju deploy github-runner small --channel latest/stable ....
+```
+
+More steps are required if you are deploying runners on OpenStack cloud.
+
+Please refer to [How to spawn OpenStack runner](how-to/openstack-runner.md) or [How to run on LXD cloud](how-to/run-on-lxd.md)
+for more information on how to deploy the runners.
+
+### MongoDB
+
+You need to deploy a MongoDB application to use as a message queue. 
+You can choose to use the machine charm or the k8s charm version, although we recommend using
+the machine charm as the k8s version may not be reachable from outside the k8s cluster.
+
+```shell
+juju switch machine-model
+juju deploy mongodb --channel 6/edge --revision 188 
+juju expose mongodb
+juju offer mongodb:database
+juju grant <user-in-k8s-cloud> consume <offer-name>
+```
+
+Integrate with the runner charms
+
+```shell
+juju integrate large mongodb
+juju integrate large-arm mongodb
+juju integrate small mongodb
+```
+
+### Define a webhook in your organisation or repository where the self-hosted runners are registered.
+
+On your repository or organisations page on Github, you need to go to the settings and create a Webhook
+(e.g. https://github.com/canonical/github-runner-operator/settings/hooks). Please make sure to select
+
+- the Webhook url to be the URL of the webhook router
+- the content type `application/json`
+- the secret you defined in the webhook router (if you have so, which is recommended for security reasons)
+- the individual event "Workflow jobs" (and only this, as all other events will just be rejected by the webhook router)
+
+### Webhook router
+
+The webhook router is a k8s charm, therefore you need to deploy it on a k8s model.
+
+First, define a routing table to decide which labels should be routed to which runner flavor
+
+```shell
+cat <<EOF > routing_table.yaml 
+- large: [large, x64]
+- large-arm: [large, arm64]
+- small: [small]
+EOF
+```
+
+We decide to route all jobs with any label combination in the set `large,x64` to the large flavor, `large,arm64` to large-arm,
+and labels with `small` to small.
+This means, depending on which labels your users are setting in the workflow file, a different runner flavor will be used to
+execute the job.
+
+
+```shell
+juju switch k8s-model
+juju deploy github-runner-webhook-router --channel latest/edge --revision 30 --config flavours=@routing_table.yaml --config default-flavour=small --config webhook-secret=<your-secret>
+juju consume <offer-url>
+juju integrate github-runner-webhook-router mongodb
+```
+
+In this example we use "small" as the default flavor, to which all jobs with empty labels (or default labels such as `self-hosted`,`linux`) 
+are routed to.
+
+
+We also need to make the webhook publicly available, so you will probably need an ingress or traefik charm to expose the webhook router.
```

Comment:
Maybe rewrite to say

> If you need to make the webhook publicly available, you will need an ingress or the traefik charm to expose the webhook router.

Reply:
The webhook needs to be publicly available to be reachable from GitHub, so I would prefer not to use a conditional ("If you need"). 

The reason I use "you will probably need" is that I don't know anything about the user's infrastructure, they may have a different solution for routing traffic to the webhook router.